### PR TITLE
Use mountPath for all log in and log out redirects

### PR DESCRIPTION
**Pull request by JeremyPlease**
Fixes ParsePlatform/parse-dashboard#604

### CODE REVIEW

**Comment by JeremyPlease** [comment]
Checking tests now...

File: src/lib/tests/e2e.js [hunk 1]
```diff
@@ -21,7 +21,8 @@ var dashboard = ParseDashboard({
       "masterKey": "masterKey",
       "appName": "MyApp"
     }
-  ]
+  ],
+  "mountPath": "/dashboard"
 });
 
 app.use('/dashboard', dashboard);
```
Comment (JeremyPlease) [inline src/lib/tests/e2e.js]: @flovilmart Can you confirm that `mountPath` should be set when mounting to a non-root path? Previously the `mountPath` was determined in the `getMount` function.

Comment (kwstasna) [inline src/lib/tests/e2e.js]: @JeremyPlease very nice work! we are waiting 🤓

**Comment by JeremyPlease** [comment]
Alrighty. Things are patched up and ready to go. 

Dashboard now properly handles being mounted as express middleware without the need to explicitly set `mountPath`.

File: Parse-Dashboard/app.js [hunk 2]
```diff
@@ -17,11 +17,12 @@ packageJson('parse-dashboard', 'latest').then(latestPackage => {
...
-function getMount(req) {
-  let url = req.url;
-  let originalUrl = req.originalUrl;
-  var mountPathLength = req.originalUrl.length - req.url.length;
-  var mountPath = req.originalUrl.slice(0, mountPathLength);
+function getMount(mountPath) {
+  // let url = req.url;
+  // let originalUrl = req.originalUrl;
+  // var mountPathLength = originalUrl.length - url.length;
+  // var mountPath = originalUrl.slice(0, mountPathLength);
```
Comment (flovilmart) [inline Parse-Dashboard/app.js]: can you remove that?

File: Parse-Dashboard/app.js [hunk 3]
```diff
@@ -59,149 +60,152 @@ module.exports = function(config, allowInsecureHTTP) {
...
-    const requestIsLocal =
-      req.connection.remoteAddress === '127.0.0.1' ||
-      req.connection.remoteAddress === '::ffff:127.0.0.1' ||
-      req.connection.remoteAddress === '::1';
-    if (!requestIsLocal && !req.secure && !allowInsecureHTTP) {
-      //Disallow HTTP requests except on localhost, to prevent the master key from being transmitted in cleartext
-      return res.send({ success: false, error: 'Parse Dashboard can only be remotely accessed via HTTPS' });
-    }
+  // wait for app to mount in order to get mountpath
+  app.on('mount', function() {
```
Comment (flovilmart) [inline Parse-Dashboard/app.js]: that is amazing! We shall use that on parse-server!

File: src/components/Sidebar/FooterMenu.react.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ import React    from 'react';
 import styles   from 'components/Sidebar/Sidebar.scss';
 
 let host = location.host.split('.');
-let urlRoot = location.protocol + '//' + host.slice(host.length - 2).join('.');
+let mountPath = window.PARSE_DASHBOARD_PATH;
```
Comment (flovilmart) [inline src/components/Sidebar/FooterMenu.react.js]: where is that set?

**Comment by flovilmart** [comment]
@JeremyPlease did you thoroughly test with the redirects etc...

File: src/components/Sidebar/FooterMenu.react.js (hunk 4 above)
Comment (JeremyPlease) [inline src/components/Sidebar/FooterMenu.react.js]: It's set in the html header returned for all pages served by express.
[line 168 of app.js](https://github.com/JeremyPlease/parse-dashboard/blob/94b46d09a80b5c38277ce3823d03cc00cdb5258a/Parse-Dashboard/app.js#L168) and
[line 193 of app.js](https://github.com/JeremyPlease/parse-dashboard/blob/94b46d09a80b5c38277ce3823d03cc00cdb5258a/Parse-Dashboard/app.js#L193)

**Comment by JeremyPlease** [comment]
@flovilmart Yeah, I tested log in, log out, and security based redirects on the following:

1. Run as CLI with no `MOUNT_PATH`
2. Run as CLI with `MOUNT_PATH=/dash`
3. Run as express middleware mounted on `/` (with and without `mountPath` in config)
4. Run as express middleware mounted on `/dashboard` (with and without `mountPath` in config)

I'm confident things work now but I'm looking forward to adding more tests soon :)

**Comment by flovilmart** [comment]
Alright then :)

**Comment by flovilmart** [comment]
Merging that, can you make the changelog / releases .21? @JeremyPlease

**Pull request merged**